### PR TITLE
Fix no-break space size

### PR DESCRIPTION
**Pull request by Dimon4eg**
Some fonts have an issue with no-break space:

- no letter definition
- not normal big width

Added test to reproduce the issue:
image

After fix:
image

### CODE REVIEW

**Comment by Dimon4eg** [comment]
[ci build]

**Comment by Dimon4eg** [comment]
[ci rebuild]

**Comment by Dimon4eg** [comment]
@minggo how to rebuild?

**Comment by drelaptop** [comment]
have rebuild CI for you

**Comment by Dimon4eg** [comment]
@drelaptop can I rebuild by self and if can then how?

**Comment by drelaptop** [comment]
AFAIK, the CI will be rebuild after a commit, and a member of cocos team have the right to rebuild too. but I didn't know the travis CI have this feature that rebuild when get a new comment like `[ci rebuild]`

any other project's CI have this feature?

**Comment by drelaptop** [comment]
[ci rebuild]

File: cocos/base/ccUTF8.cpp [hunk 1]
```diff
@@ -50,10 +50,10 @@ std::string format(const char* format, ...)
     va_end(args);
 
     if (nret >= 0) {
-        if (nret < buffer.length()) {
+        if (nret < (int)buffer.length()) {
             buffer.resize(nret);
         }
-        else if (nret > buffer.length()) { // VS2015/2017 or later Visual Studio Version
+        else if (nret > (int)buffer.length()) { // VS2015/2017 or later Visual Studio Version
```
Comment (drelaptop) [inline cocos/base/ccUTF8.cpp]: it's not a good type conversion when compare `int` and `unsigned int`, convert `unsigned ` to `signed` might  cause negative value. when the int value `nret >= 0`, convert it to `signed` always right.

so I think `(unsigned int)nret > buffer.length()` is a better choice

**Review by drelaptop — CHANGES_REQUESTED** [review]

**Comment by drelaptop** [comment]
Thanks for this good patch @Dimon4eg. It works well when I tested on macOS and Android, little suggestion have commented, hope your feedback.

**Comment by Dimon4eg** [comment]
Corrected, thank you @drelaptop

**Comment by drelaptop** [comment]
review and test pass, please merge @minggo

**Pull request merged**